src/lib/repositories/impl_oauth2/oauth2_repository_impl.py

```diff
@@ -0,0 +1,257 @@
+import bcrypt
+from injector import inject
+from sqlalchemy import text
+from sqlalchemy.engine.base import Engine
+import jwt
+
+from src.constants.http_status_code import HttpStatus
+from src.constants.oauth2 import CredentialType
+from src.env_config import get_env_config_instance
+from src.utils.oauth2_util import create_token
+from src.utils.sql_oath_queries import (
+    SQL_QUERY_TO_GET_USER_BY_USER_NAME,
+    SQL_QUERY_TO_GET_CLIENT_BY_CLIENT_ID,
+    SQL_QUERY_TO_GET_CLIENT_BY_ID,
+    SQL_QUERY_TO_ADD_ACCESS_TOKEN,
+    SQL_QUERY_TO_ADD_REFRESH_TOKEN,
+    SQL_QUERY_TO_GET_REFRESH_TOKEN_BY_TOKEN,
+    SQL_QUERY_TO_GET_ACCESS_TOKEN_BY_TOKEN,
+    SQL_QUERY_TO_GET_REFRESH_TOKEN_BY_ID,
+    SQL_QUERY_TO_DELETE_ACCESS_TOKEN,
+    SQL_QUERY_TO_DELETE_REFRESH_TOKEN,
+    SQL_QUERY_TO_GET_SCOPE_BY_CLIENT_ID,
+)
+
+
+class Oauth2Repository:
+    @inject
+    def __init__(self, engine: Engine):
+        self.env_config = get_env_config_instance()
+        self.engine = engine
+
+    def login(
+        self, client_id, client_secret, credential_type, user_name=None, password=None
+    ):
+        client = self._get_client_by_client_id(client_id)
+
+        if client.client_secret == client_secret:
+
+            if credential_type == CredentialType.CLIENT_CREDENTIAL:
+                tokens = self._client_credential_login(client)
+
+                return tokens
+
+            else:
+                tokens = self._client_user_credential_login(client, user_name, password)
+                return tokens
+        return HttpStatus.UNAUTHORIZED.value
```

Comment:
Don't do this, if you a situation like this one prefer raising an error saying unsupported credentials type